GVFS/GVFS.Common/GVFSEnlistmentHealthCalculator.cs

```diff
@@ -0,0 +1,257 @@
+    /// of a gvfs enlistment. Constructed with the lists of paths for the enlistment, and then
+    /// internally stores the calculated information. Compute or recompute via CalculateStatistics
+    /// with an optional parameter to only look for paths which are under the specified directory
+    /// </summary>
+    public class GVFSEnlistmentHealthCalculator
+    {
+        // In the context of this class, hydrated files are placeholders or modified paths
+        // The total number of hydrated files is this.PlaceholderCount + this.ModifiedPathsCount
+        private readonly GVFSEnlistmentPathData enlistmentPathData;
+
+        public GVFSEnlistmentHealthCalculator(GVFSEnlistmentPathData pathData)
+        {
+            this.enlistmentPathData = pathData;
+        }
+
+        public GVFSEnlistmentHealthData CalculateStatistics(string parentDirectory)
+        {
+            int gitTrackedItemsCount = 0;
+            int placeholderCount = 0;
+            int modifiedPathsCount = 0;
+            Dictionary<string, int> gitTrackedItemsDirectoryTally = new Dictionary<string, int>(StringComparer.OrdinalIgnoreCase);
+            Dictionary<string, int> hydratedFilesDirectoryTally = new Dictionary<string, int>(StringComparer.OrdinalIgnoreCase);
+
+            // Parent directory is a path relative to the root of the repository which is already in git format
+            if (!parentDirectory.EndsWith(GVFSConstants.GitPathSeparatorString) && parentDirectory.Length > 0)
+            {
+                parentDirectory += GVFSConstants.GitPathSeparator;
+            }
+
+            if (parentDirectory.StartsWith(GVFSConstants.GitPathSeparatorString))
+            {
+                parentDirectory = parentDirectory.TrimStart(GVFSConstants.GitPathSeparator);
+            }
+
+            gitTrackedItemsCount += this.CategorizePaths(this.enlistmentPathData.GitFolderPaths, gitTrackedItemsDirectoryTally, parentDirectory);
+            gitTrackedItemsCount += this.CategorizePaths(this.enlistmentPathData.GitFilePaths, gitTrackedItemsDirectoryTally, parentDirectory);
+            placeholderCount += this.CategorizePaths(this.enlistmentPathData.PlaceholderFolderPaths, hydratedFilesDirectoryTally, parentDirectory);
+            placeholderCount += this.CategorizePaths(this.enlistmentPathData.PlaceholderFilePaths, hydratedFilesDirectoryTally, parentDirectory);
+            modifiedPathsCount += this.CategorizePaths(this.enlistmentPathData.ModifiedFolderPaths, hydratedFilesDirectoryTally, parentDirectory);
+            modifiedPathsCount += this.CategorizePaths(this.enlistmentPathData.ModifiedFilePaths, hydratedFilesDirectoryTally, parentDirectory);
+
+            Dictionary<string, decimal> mostHydratedDirectories = new Dictionary<string, decimal>();
+
+            // Map directory names to the corresponding health data from gitTrackedFilesDirectoryTally and hydratedFilesDirectoryTally

```

Comment:
Is this comment out of date?  What is `gitTrackedFilesDirectoryTally`?

Reply:
Yeah the comment is outdated, somehow the refactoring must have missed it. It should say `// Map directory names to the corresponding health data from gitTrackedItemsDirectoryTally and hydratedFilesDirectoryTally`

Reply:
Adding this to issue: https://github.com/microsoft/VFSForGit/issues/1298